otter/integration/lib/test_autoscale.py

```diff
@@ -98,3 +100,83 @@ def test_poll_until_timeout(self):
             self.assertNoResult(d)
         self.clock.advance(10)
         self.failureResultOf(d, TimedOutError)
+
+
+class GetServicenetIPs(SynchronousTestCase):
+    """
+    Tests for :func:`ScalingGroup.get_servicenet_ips`.
+    """
+    def setUp(self):
+        """Create our simulated clock and scaling group."""
+        self.clock = Clock()
+        self.pool = object()
+        self.queried_server_ids = []
+
+        class FakeRCS(object):
+            endpoints = {'nova': 'novaurl'}
+            token = "token"
+
+        @attributes(['code', 'fake_json_response'])
+        class Resp(object):
+            pass
+
+        class FakeTreq(object):
```

Comment:
@manishtomar It's true; I have pressure from Pranav (and not without merit) to keep as much of this library independent of Otter as possible, because we'd like to take this work and apply it eventually to other projects in the office.  Also, the work being invested in developing this Trial-based library will eventually make its way into OpenCafe.  So, it's very important I distinguish between what is absolutely required to be reused from Otter versus what I can reasonably implement myself externally.

That means that, eventually, I'll need to implement my own authentication mechanism in Identity class.  But, I'll cross that bridge when I get there.

And, since it's so early in the morning for me still, when I say "I", I of course mean the QE group as a whole.  :)